### PR TITLE
TRAIN-216: Create User Controller

### PR DESCRIPTION
**Pull request by r-barlow**

### CODE REVIEW

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import za.co.neslotech.spring.trainspringbootinsight.model.User;
+import za.co.neslotech.spring.trainspringbootinsight.repository.IUserRepository;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController extends AGenericController<User, Long>{
+
+    @Autowired
+    private IUserRepository userRepository;
```
Comment (JOlsen-Neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java]: Following the pattern of marking everything final that can be final, I would mark this as a final variable and then initialise it in the constructor. The Autowiring will occur automatically

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import za.co.neslotech.spring.trainspringbootinsight.model.User;
+import za.co.neslotech.spring.trainspringbootinsight.repository.IUserRepository;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController extends AGenericController<User, Long>{
```
Comment (JOlsen-Neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java]: Please make sure you format your files

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import za.co.neslotech.spring.trainspringbootinsight.service.UserService;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController extends CrudController<User, Long> {
+
+    private final UserService service;
+
+    @Autowired
+    public UserController(UserService service) {
```
Comment (JOlsen-Neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java]: Mark this as final

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    private final IUserRepository repository;
+
+    @Autowired
+    public UserService(final IUserRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public User update(final User entity) {
```
Comment (JOlsen-Neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/service/UserService.java:25]: Shouldn't there be a create method as well?

**Review by JOlsen-Neslo — CHANGES_REQUESTED** [review]

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import za.co.neslotech.spring.trainspringbootinsight.service.CrudService;
+import za.co.neslotech.spring.trainspringbootinsight.service.UserService;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController extends CrudController<User, Long> {
+
+    private final UserService service;
+
+    @Autowired
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/controller/UserController.java]: With the new version of spring, marking the constructor of a controller, service or repository annotated bean is redundant.

**Review by JOlsen-Neslo — APPROVED** [review]

**Review by colsen-neslo — APPROVED** [review]

**Pull request merged**